pkg/query-service/app/logs/v3/query_builder.go

```diff
@@ -0,0 +1,349 @@
+package v3
+
+import (
+	"fmt"
+	"math"
+	"strings"
+
+	"go.signoz.io/signoz/pkg/query-service/constants"
+	v3 "go.signoz.io/signoz/pkg/query-service/model/v3"
+	"go.signoz.io/signoz/pkg/query-service/utils"
+)
+
+var aggregateOperatorToPercentile = map[v3.AggregateOperator]float64{
+	v3.AggregateOperatorP05:         0.05,
+	v3.AggregateOperatorP10:         0.10,
+	v3.AggregateOperatorP20:         0.20,
+	v3.AggregateOperatorP25:         0.25,
+	v3.AggregateOperatorP50:         0.50,
+	v3.AggregateOperatorP75:         0.75,
+	v3.AggregateOperatorP90:         0.90,
+	v3.AggregateOperatorP95:         0.95,
+	v3.AggregateOperatorP99:         0.99,
+	v3.AggregateOperatorHistQuant50: 0.50,
+	v3.AggregateOperatorHistQuant75: 0.75,
+	v3.AggregateOperatorHistQuant90: 0.90,
+	v3.AggregateOperatorHistQuant95: 0.95,
+	v3.AggregateOperatorHistQuant99: 0.99,
+}
+
+var aggregateOperatorToSQLFunc = map[v3.AggregateOperator]string{
+	v3.AggregateOperatorAvg:     "avg",
+	v3.AggregateOperatorMax:     "max",
+	v3.AggregateOperatorMin:     "min",
+	v3.AggregateOperatorSum:     "sum",
+	v3.AggregateOperatorRateSum: "sum",
+	v3.AggregateOperatorRateAvg: "avg",
+	v3.AggregateOperatorRateMax: "max",
+	v3.AggregateOperatorRateMin: "min",
+}
+
+var logsOperatorsEnabled = map[v3.FilterOperator]struct{}{
+	v3.FilterOperatorEqual:           {},
+	v3.FilterOperatorNotEqual:        {},
+	v3.FilterOperatorLessThan:        {},
+	v3.FilterOperatorLessThanOrEq:    {},
+	v3.FilterOperatorGreaterThan:     {},
+	v3.FilterOperatorGreaterThanOrEq: {},
+	v3.FilterOperatorLike:            {},
+	v3.FilterOperatorNotLike:         {},
+	v3.FilterOperatorRegex:           {},
+	v3.FilterOperatorNotRegex:        {},
+	v3.FilterOperatorIn:              {},
+	v3.FilterOperatorNotIn:           {},
+	// (todo) check contains/not contains/exists/not exists
```

Comment:
These needs to be supported as well.

Reply:
I didn't find any difference between `contains/not contains` and `like/not like`. Even in metrics implementation, it's the same.

In logs previously we converted contains to -> `ilike`. I am not really sure what should we do.

Reply:
I didn't have the `contains/not contains` earlier, so I imagined this was important for logs/traces. The difference, I believe, is `contains` is regular text, but the `like/not like` is an expression in the format which includes `%`, `_` for matching. For example, `status_code` like `2__` matches the `2xx`, but that would differ from the `status_code` contains `2`.

Reply:
Have added contains,ncontains,exists,nexists